eland/operations.py

```diff
@@ -1509,6 +1571,49 @@ def batch_size() -> None:
     def show_progress(self) -> bool:
         return self._show_progress
 
+class PandasDataFrameIterator(ABC):
+    def __init__(self, query_compiler, es_results_generator, post_processing):
+        self._query_compiler = query_compiler
+        self._es_results_generator = es_results_generator
+        self._post_processing = post_processing
+
+    def __iter__(self):
+        return self
+
+    @abstractmethod
+    def __next__(self):
+        pass
+
+    def _es_results_to_pandas(self, es_results):
+        _, df = self._query_compiler._es_results_to_pandas(es_results)
+        df = Operations._apply_df_post_processing(df, self._post_processing)
+        return df
+
+class PandasDataFrameRowsIterator(PandasDataFrameIterator):
```

Comment:
Pandas returns generator right. Shall we do the same ?

Reply:
I think we can all return `generator`, I have completed the modification